engine/lite/k8sDeploymentManager/src/test/scala/pl/touk/nussknacker/k8s/manager/deployment/DeploymentPreparerTest.scala

```diff
@@ -0,0 +1,131 @@
+    deployment.spec.get.template.metadata.labels.keys should contain(K8sDeploymentManager.scenarioVersionLabel)
+
+    deployment.spec.get.template.spec.get.volumes should contain(Volume("configmap", Volume.ConfigMapVolumeSource("fooConfigMap")))
+    deployment.spec.get.template.spec.get.containers.head.name shouldBe "runtime"
+    deployment.spec.get.template.spec.get.containers.head.image should startWith("touk/nussknacker-lite-kafka-runtime")
+    deployment.spec.get.template.spec.get.containers.head.env should contain(EnvVar("SCENARIO_FILE", StringValue("/data/scenario.json")))
+    deployment.spec.get.template.spec.get.containers.head.env should contain(EnvVar("CONFIG_FILE", StringValue("/opt/nussknacker/conf/application.conf,/data/modelConfig.conf")))
+    deployment.spec.get.template.spec.get.containers.head.env should contain(EnvVar("POD_NAME", FieldRef("metadata.name")))
+    deployment.spec.get.template.spec.get.containers.head.volumeMounts.head.name shouldBe "configmap"
+    deployment.spec.get.template.spec.get.containers.head.volumeMounts.head.mountPath shouldBe "/data"
+
+  }
+
+
+  test("should prepare deployment when k8sDeploymentConfig is provided") {
+    import scala.jdk.CollectionConverters.seqAsJavaListConverter
+    val config = K8sDeploymentManagerConfig(k8sDeploymentConfig =
+      ConfigFactory.empty()
+        .withValue("metadata.name", ConfigValueFactory.fromAnyRef("shouldBeOverriden"))
+        .withValue("metadata.labels.my-label", ConfigValueFactory.fromAnyRef("my-key"))
+        .withValue("metadata.annotations.my-label", ConfigValueFactory.fromAnyRef("my-key"))
+        .withValue("spec.replicas", ConfigValueFactory.fromAnyRef(3))
+        .withValue("spec.minReadySeconds", ConfigValueFactory.fromAnyRef(3))
+        .withValue("spec.progressDeadlineSeconds", ConfigValueFactory.fromAnyRef(3))
+        .withValue("spec.strategy", ConfigValueFactory.fromAnyRef("RollingUpdate"))
+        .withValue("spec.selector.matchLabels.override", ConfigValueFactory.fromAnyRef("shouldBeOverriden"))
+        .withValue("spec.template.metadata.name", ConfigValueFactory.fromAnyRef("shouldBeOverriden"))
+        .withValue("spec.template.metadata.labels.my-label", ConfigValueFactory.fromAnyRef("my-key"))
+        .withValue("spec.template.spec.volumes", ConfigValueFactory.fromIterable(List(
+          ConfigFactory.empty()
+            .withValue("name", ConfigValueFactory.fromAnyRef("my-volume")).root()
+        ).asJava))
+        .withValue("spec.template.spec.containers", ConfigValueFactory.fromIterable(List(
+          ConfigFactory.empty()
+            .withValue("name", ConfigValueFactory.fromAnyRef("my-container"))
+            .withValue("image", ConfigValueFactory.fromAnyRef("my-image"))
+            .root()
+        ).asJava))
+    )
+    val deployment = DeploymentPreparer(ProcessVersion.empty, config, configMapId)
+
+    deployment.metadata.name shouldBe "scenario-1-x"
```

Comment:
I wonder, would equality based assertion on whole deployment be better?

Reply:
I agree with that - take a look at `labels.keys` verification - you don't check if something more is added. It was supposed to be like that?. 
Also would be nice to reference to given values in expected value.

Reply:
done